backend/api.py

```
@@ -379,6 +387,206 @@ def get_molecule_3d(index: int):
+        
+        functional_groups = identify_functional_groups(molecule['smiles'])
+        
+        return jsonify({
+            'success': True,
+            'functional_groups': functional_groups,
+            'smiles': molecule['smiles']
+        })
+    
+    except ValueError as e:
+        return jsonify({'success': False, 'error': str(e)}), 400
+    except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)